project/CelebornBuild.scala

```diff
@@ -753,7 +753,13 @@ trait SparkClientProjects {
   
         (assembly / assemblyMergeStrategy) := {
           case m if m.toLowerCase(Locale.ROOT).endsWith("manifest.mf") => MergeStrategy.discard
-          case "META-INF/LICENSE" | "META-INF/NOTICE" => MergeStrategy.preferProject
+          // the LicenseAndNoticeMergeStrategy always picks the license/notice file from the current project
+          case m @ ("META-INF/LICENSE" | "META-INF/NOTICE") => CustomMergeStrategy("LicenseAndNoticeMergeStrategy") { conflicts =>
+            val entry = conflicts.head
```

Comment:
is this way reliable?

Reply:
seems we can define a custom MergeStragegy class to eliminate the duplication

Reply:
> is this way reliable?
The strategy takes effect only when the conflict files are non-empty. The built-in strategy `MergeStrategy.first` encompasses an identical function call

https://github.com/sbt/sbt-assembly/blob/64dedc8f6f829fbe5feca9d59061a5e344419bd2/src/main/scala/sbtassembly/MergeStrategy.scala#L71-L73

Reply:
> seems we can define a custom MergeStragegy class to eliminate the duplication

It's hard to do, I need to conduct more in-depth research.

```
[error] `value` can only be used within a task or setting macro, such as :=, +=, ++=, Def.task, or Def.setting.
[error]     val projectLicenseFile = (Compile / resourceDirectory).value / entry.target
```

Reply:
I'm not sure if the `conflicts.head` always returns the current project file(if provided), but the test result looks good. please at least document something above `conflicts.head`

Reply:
> I'm not sure if the `conflicts.head` always returns the current project file(if provided)
no matter what the `conflicts.head` returns, the content of LICENSE/NOTICE will always be read from the current project resource directory.